vllm/transformers_utils/tokenizers/baichuan.py

```diff
@@ -148,9 +148,9 @@ def save_vocabulary(self,
             `Tuple(str)`: Paths to the files saved.
         """
         if not os.path.isdir(save_directory):
-            logger.error(f"Vocabulary path ({save_directory}) "
-                         "should be a directory")
-            return
+            raise ValueError(f"Vocabulary path ({save_directory}) "
```

Comment:
QQ: what's the motivation behind the functional change here?

Reply:
This function was just copied in from some `baichuan` repo. The exception is more of a this should never happen thing. `save_vocabulary` is not used by vllm. I also don't want to remove the file entirely - but the `return None` is not compliant to the function signature.

Reply:
*file -> function

I think the reason why the function is included at all, is to keep the diff to baichuan repo low.